File: java/src/main/java/org/nc/operational/TimePeriod.java
package org.nc.operational;

import java.time.LocalDate;

public class TimePeriod {
    private final LocalDate startDate;
    private final LocalDate endDate;

    public TimePeriod(LocalDate startDate, LocalDate endDate) {
        this.startDate = startDate;
        this.endDate = endDate;
    }

    public LocalDate getStartDate() {
        return startDate;
    }

    public LocalDate getEndDate() {
        return endDate;
    }

    public boolean includes(LocalDate date) {
        if (date == null) return false;
        boolean afterStart = (startDate == null) || !date.isBefore(startDate);
        boolean beforeEnd = (endDate == null) || !date.isAfter(endDate);
        return afterStart && beforeEnd;
    }
}

File: python/operational/TimePeriod.py
from datetime import date
from typing import Optional


class TimePeriod:
    def __init__(self, start: Optional[date], end: Optional[date]):
        self.start = start
        self.end = end

    def get_start_date(self) -> Optional[date]:
        return self.start

    def get_end_date(self) -> Optional[date]:
        return self.end

    def includes(self, check_date: date) -> bool:
        if check_date is None: return False

        after_start = self.start is None or check_date >= self.start
        before_end = self.end is None or check_date <= self.end
        
        return after_start and before_end